File: wiki/versions/1.19/item/simple_item/scala/src/main/scala/org/quiltmc/wiki/simple_item/SimpleItemExample.scala
package org.quiltmc.wiki.simple_item

import net.minecraft.item.{Item, ItemGroup}
import net.minecraft.util.Identifier
import net.minecraft.util.registry.Registry
import org.quiltmc.loader.api.ModContainer
import org.quiltmc.qsl.base.api.entrypoint.ModInitializer
import org.quiltmc.qsl.item.setting.api.QuiltItemSettings

// Can be made an object with a language adapter for scala
class SimpleItemExample extends ModInitializer {

    override def onInitialize(mod: ModContainer): Unit = {
        // @start Registration
        Registry.register(Registry.ITEM, Identifier(mod.metadata.id, "example_item"), SimpleItemExample.EXAMPLE_ITEM)
        // @end Registration
    }
}
object SimpleItemExample {

    // @start Declaration
    final val EXAMPLE_ITEM: Item = Item(QuiltItemSettings().group(ItemGroup.MISC))
    // @end Declaration
}

File: wiki/versions/1.19/item/simple_item/java/src/main/java/org/quiltmc/wiki/simple_item/SimpleItemExample.java
package org.quiltmc.wiki.simple_item;

import net.minecraft.item.Item;
import net.minecraft.item.ItemGroup;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;
import org.quiltmc.loader.api.ModContainer;
import org.quiltmc.qsl.base.api.entrypoint.ModInitializer;
import org.quiltmc.qsl.item.setting.api.QuiltItemSettings;

public class SimpleItemExample implements ModInitializer {

	// @start Declaration
	public static final Item EXAMPLE_ITEM = new Item(new QuiltItemSettings().group(ItemGroup.MISC));
	// @end Declaration

	@Override
	public void onInitialize(ModContainer mod) {
		// @start Registration
		Registry.register(Registry.ITEM, new Identifier(mod.metadata().id(), "example_item"), EXAMPLE_ITEM);
		// @end Registration
	}
}
